src/codemagic/tools/_app_store_connect/actions/publish_action.py

```diff
@@ -148,6 +154,9 @@ def _submit_build_to_testflight(self, build: Build, app: App, max_processing_min
         build = self._wait_until_build_is_processed(build, max_processing_minutes)
         self.create_beta_app_review_submission(build.id)
 
+    def _submit_build_to_beta_groups(self, build, beta_group_names):
```

Comment:
Why the proxy method. Wherever you call `self. _submit_build_to_beta_groups(build, group_name)` you could just as well do `self.add_build`.